03-Custom-Actions-In-Python/06-Improving-Performance.ipynb

```diff
@@ -0,0 +1,276 @@
+{
```

Comment:
<p>I think these general guidelines are a bit too general to start off with. You talk about the importance of profiling, but don't demonstrate it. The same goes for scipy, cython, numba, etc...</p><p> </p><p>I would invert the order these topics are presented and start by mentioning how the local snapshots avoid copy and gather overheads. Then explain that the local snapshots interoperate through numpy/array interfaces with scipy and other packages in the scientific python ecosytem.</p><p> </p><p>Then, if you want, you can note that higher performance may be achievable with compiled solutions such as numpy, cython, or C++.</p><p> </p><p>If you don't find it too long, you could demonstrate profiling by comparing the run time of simulations using the snapshot based updater vs the local snapshot based one. Are there good profiling tools that work inside jupyter notebooks that would work for profiling HOOMD operations?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/hoomd-examples/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Reply:
<p>We could use snakeviz</p>